yt/frontends/ramses/data_structures.py

```diff
@@ -197,13 +214,95 @@ def fill(self, fd, fields, selector, file_handler):
         # Initializing data container
         for field in fields:
             tr[field] = np.zeros(cell_count, 'float64')
-
         fill_hydro(fd, file_handler.offset,
-                   file_handler.level_count, levels, cell_inds,
+                   file_handler.level_count,
+                   [self.domain_id-1],
+                   levels, cell_inds,
                    file_inds, ndim, all_fields, fields, tr,
                    oct_handler)
         return tr
 
+    def _fill_with_ghostzones(self, fd, fields, selector, file_handler, num_ghost_zones):
+        ndim = self.ds.dimensionality
+        ncpu = self.ds.parameters['ncpu']
+        # Here we get a copy of the file, which we skip through and read the
+        # bits we want.
+        oct_handler = self.oct_handler
+        all_fields = [f for ft, f in file_handler.field_list]
+        fields = [f for ft, f in fields]
+        tr = {}
+
+        cell_count = selector.count_octs(self.oct_handler, self.domain_id) * self.nz**ndim
+
+        gz_cache = getattr(self, "_ghost_zone_cache", None)
+        if gz_cache:
+            levels, cell_inds, file_inds, domains = gz_cache
+        else:
+            gz_cache = levels, cell_inds, file_inds, domains = self.oct_handler.file_index_octs_with_ghost_zones(
+            selector, self.domain_id, cell_count)
+            self._ghost_zone_cache = gz_cache
+
+        # Initializing data container
+        for field in fields:
+            tr[field] = np.zeros(cell_count, 'float64')
+        fill_hydro(fd, file_handler.offset,
+                   file_handler.level_count,
+                   list(range(ncpu)),
+                   levels, cell_inds,
+                   file_inds, ndim, all_fields, fields, tr,
+                   oct_handler,
+                   domains=domains)
+        return tr
+
+    @property
+    def fwidth(self):
+        fwidth = super(RAMSESDomainSubset, self).fwidth
+        if self._num_ghost_zones > 0:
+            fwidth = super(RAMSESDomainSubset, self).fwidth.reshape(-1, 8, 3)
```

Comment:
```suggestion
            fwidth = fwidth.reshape(-1, 8, 3)
```

Reply:
I'm addressing this one on #2425 where it first appeared.